src/API/Site/Controllers/MerchantCenter/ShippingRateController.php

```diff
@@ -207,6 +207,9 @@ protected function get_create_rate_callback(): callable {
 			$update_query = $this->create_query();
 			$country_code = $request->get_param( 'country_code' );
 
+			// First we delete all rates for this country (to avoid duplicates).
+			$this->create_query()->delete( 'country', $country_code );
```

Comment:
Is it not better to try to update instead of always deleting and then inserting? The previous code checked for a matching country and method, whereas this code deletes all for the same country. I understand we don't have multiple methods per country (yet), but it might be something to keep in mind if we are possibly going to change this logic later on.

Reply:
Hmm. You're right there's no need to delete them now that there's only one rate per country and method. We can get back to this later. I originally modified this because I had a different structure in mind for storing the shipping classes but now that they are tied to each country+method, we can prevent duplicates by just checking those keys.

Reply:
Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/1175/commits/6c962241ce4ac72193139be3d0fd78f48378d483